modular_skyrat/modules/assault_operatives/code/interrogator.dm

```diff
@@ -0,0 +1,218 @@
+	playsound(loc, 'sound/items/rped.ogg', 60)
+	say("Stage one complete!")
+	minor_announce("SECURITY BREACH DETECTED, NETWORK COMPROMISED! READING COORDINATES...", "GoldenEye Defence Network")
+	timer_id = addtimer(CALLBACK(src, .proc/stage_two), rand(STAGE_PROCESS_TIME_LOWER, STAGE_PROCESS_TIME_UPPER), TIMER_STOPPABLE|TIMER_UNIQUE)
+
+/obj/machinery/interrogator/proc/stage_two()
+	if(!check_requirements())
+		say("Critical error! Aborting.")
+		playsound(src, 'sound/machines/scanbuzz.ogg', 100)
+		return
+	to_chat(human_occupant, span_userdanger("You feel a sharp pain as a drill penetrates your skull, it's unbearable!"))
+	human_occupant.emote("scream")
+	human_occupant.apply_damage(30, BRUTE, BODY_ZONE_HEAD)
+	playsound(src, 'sound/effects/wounds/blood1.ogg', 100)
+	playsound(src, 'sound/items/drill_use.ogg', 100)
+	say("Stage two complete!")
+	minor_announce("SECURITY BREACH DETECTED, NETWORK COMPROMISED! COORDINATES: [x], [y], [z]", "GoldenEye Defence Network")
+	timer_id = addtimer(CALLBACK(src, .proc/stage_three), rand(STAGE_PROCESS_TIME_LOWER, STAGE_PROCESS_TIME_UPPER), TIMER_STOPPABLE|TIMER_UNIQUE)
+
+/obj/machinery/interrogator/proc/stage_three()
+	if(!check_requirements())
+		say("Critical error! Aborting.")
+		playsound(src, 'sound/machines/scanbuzz.ogg', 100)
+		return
+	to_chat(human_occupant, span_userdanger("You feel something penetrating your brain, it feels as though your childhood memories are fading! Please, make it stop! After a moment of slience you realise you can't remember what happened to you!"))
+	human_occupant.emote("scream")
+	human_occupant.apply_damage(20, BRUTE, BODY_ZONE_HEAD)
+	human_occupant.Jitter(3 MINUTES)
+	human_occupant.Unconscious(1 MINUTES)
+	playsound(src, 'sound/effects/dismember.ogg', 100)
+	playsound(src, 'sound/machines/ping.ogg', 100)
+	say("Process complete! A key is being sent aboard! Crew will shortly detect the keycard!")
+	send_keycard()
+	processing = FALSE
+	locked = FALSE
+	update_appearance()
+	addtimer(CALLBACK(src, .proc/announce_creation), ALERT_CREW_TIME)
+
+/obj/machinery/interrogator/proc/announce_creation()
+	priority_announce("CRITICAL SECURITY BREACH DETECTED! A GoldenEye authentication key has been illegally printed, locate it at all costs!", "GoldenEye Defence Network", ANNOUNCER_KLAXON)
+	for(var/obj/item/pinpointer/nuke/disk_pinpointers in GLOB.pinpointer_list)
+		disk_pinpointers.switch_mode_to(TRACK_GOLDENEYE) //Pinpointer will track the newly created goldeneye key.
+
+/obj/machinery/interrogator/proc/send_keycard()
+	var/turf/landingzone = find_drop_turf()
+	var/obj/item/goldeneye_key/new_key
+	if(!landingzone)
+		new_key = new(src)
+	else
+		new_key = new
+	new_key.extract_name = human_occupant.real_name
+	var/datum/objective/interrogate/objective = get_goldeneye_target(human_occupant)
+	objective.linked_key = src
+	new_key.linked_objective = objective
+	var/obj/structure/closet/supplypod/pod = new
```

Comment:
The supply pod doesn't have a location... Unless that's somehow normal, in which case, don't mind me.

Reply:
Ye, it's normal, it starts in nullspace